x/model/types/messages_model.go

```diff
@@ -106,7 +108,7 @@ func (msg *MsgCreateModel) ValidateBasic() error {
 		return err
 	}
 
-	if msg.EnhancedSetupFlowOptions == 0 {
+	if msg.EnhancedSetupFlowOptions&1 == 1 {
```

Comment:
Do we have a logic and tests that EnhancedSetupFlow fields must be present IF AND ONLY IF EnhancedSetupFlowOptions&1 == 1 when Creating a Model?

Reply:
Added tests